internal/armada/repository/event_nats.go

```diff
@@ -76,13 +77,14 @@ func (p *RedisEventProcessor) handleBatch(batch []*eventstream.Message) error {
 
 	err := p.repository.ReportEvents(events)
 	if err != nil {
-		return fmt.Errorf("error while reporting events to event store for %d events: %v", len(events), err)
+		return fmt.Errorf("[handleBatch] error reporting %d events: %w", len(events), err)
 	}
 
 	for _, msg := range batch {
 		err = msg.Ack()
 		if err != nil {
-			log.Errorf("error while acknowledging event: %v", err)
+			err = fmt.Errorf("[handleBatch] error acknowledging event: %s", err)
```

Comment:
Yes. Errors wrapped with `%w` effectively become part of the API since they can be extracted programmatically. Hence, I've avoided using `%w` to wrap errors coming from libraries, since those could change, which would change the API. Since this is internal code, I don't think it matters much, but that's at least the motivation :)